net-vpn/i2p/i2p-0.9.42.ebuild

```diff
@@ -146,12 +153,59 @@ src_install() {
 	newinitd "${FILESDIR}/i2p.init" i2p
 	systemd_dounit "${FILESDIR}/i2p.service"
 
+	# setup log
+	keepdir /var/log/i2p
+	fowners i2p:i2p /var/log/i2p
+
 	# setup user
-	keepdir '/var/lib/i2p/.i2p'
-	fowners i2p:i2p '/var/lib/i2p/.i2p'
+	keepdir /var/lib/i2p
+	keepdir /var/lib/i2p/app
+	keepdir /var/lib/i2p/router
+	keepdir /var/lib/i2p/config
+	fowners i2p:i2p /var/lib/i2p
 }
 
 pkg_postinst() {
-	elog "Custom configuration belongs in ${EPREFIX}/var/lib/i2p/.i2p to avoid being overwritten."
-	elog 'I2P can be configured through the web interface at http://localhost:7657/console'
+	local old_i2pdir="${EPREFIX}/var/lib/i2p/.i2p" new_i2pdir="${EPREFIX}/var/lib/i2p"
+
+	[ -e "${old_i2pdir}" ] || return
+
+	elog "User is now delegated to acct-user, ${new_i2pdir} is split"
+	elog "into subdirs. It will now try to split ${old_i2pdir} accordingly."
+
+	migrate() {
+		local dest="${1}"
+		shift
+
+		local ret=true
+		for src
+		do
+			[ -e "${src}" ] || continue
+			mv "${src}" "${dest}" || ret=false
+		done
+
+		$ret
+	}
+
+	ebegin 'Migrating'
+	local ret=0
+	chown -R i2p:i2p "${EPREFIX}/var/lib/i2p" || ret=1
```

Comment:
Normally you'd want to add `|| die` here, but this doesnt seem to be needing to be fatal.

Reply:
Correct; as I try to progress most of the migration before failing, `ret` is used to mark failure.